uda_udf_test.py

```diff
@@ -0,0 +1,130 @@
+from typing import Tuple, NamedTuple, Callable
+
+from sdcm.cluster import BaseNode
+from sdcm.send_email import LongevityEmailReporter
+from sdcm.stress_thread import CassandraStressThread
+from sdcm.tester import ClusterTester
+from sdcm.utils.udf import UDFS
+from sdcm.utils.uda import UDAS
+
+
+class UDVerification(NamedTuple):
+    name: str
+    query: str
+    verifier_func: Callable
+
+
+class UDAUDFTest(ClusterTester):
+    """
+    Test Scylla with User Defined Functions and User Defined Aggregates,
+    using cassandra-stress.
+    """
+    KEYSPACE_NAME = "ks"
+    CF_NAME = "uda_udf"
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.email_reporter = LongevityEmailReporter(email_recipients=self.params.get('email_recipients'),
+                                                     logdir=self.logdir)
+
+    def test_uda_and_udf(self) -> None:
+        self.log.info("Starting UDA/UDF test...")
+        self.prewrite_db_with_data()
+        node: BaseNode = self.db_cluster.get_node()
+
+        for udf in UDFS.values():
+            self.log.info("Creating the following UDF: %s", udf.name)
+            cmd = udf.get_create_query(ks="ks")
+            with self.db_cluster.cql_connection_patient(node=node) as session:
+                session.execute(cmd)
+            self.log.info("UDF %s created.", udf.name)
+
+        self._verify_udf_functions()
+
+        for uda in UDAS.values():
+            cmd = uda.get_create_query_string(ks="ks")
+            node.run_cqlsh(cmd=cmd)
+            self.log.info("UDA %s created.", uda.name)
+
+        self._verify_uda_aggregates()
+
+        uda_udf_thread = self.run_uda_udf_thread()
+
+        # wait for stress to complete
+        self.verify_stress_thread(cs_thread_pool=uda_udf_thread)
+        self.log.info("Test completed")
+
+    def run_uda_udf_thread(self) -> Tuple[CassandraStressThread, CassandraStressThread]:
```

Comment:
Not sure why returned type annotation has 2 threads while returning a single thread

Reply:
Good catch - fixed